include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -1588,6 +1588,252 @@ namespace internal
+            &                            accessor,
+          const unsigned int             vertex,
+          const unsigned int             d,
+          const types::global_dof_index &index_value,
+          const unsigned int             fe_index)
+        {
+          accessor.set_vertex_dof_index(vertex, d, index_value, fe_index);
+        }
+
+        /**
+         * Set DoF index for lines, quads, and inner degrees of freedom.
+         */
+        template <int structdim_>
+        static DEAL_II_ALWAYS_INLINE void
+        process_dof(const dealii::DoFAccessor<structdim_,
+                                              DoFHandlerType,
+                                              level_dof_access> &accessor,
+                    const unsigned int                           d,
+                    const types::global_dof_index &              index_value,
+                    const unsigned int                           fe_index)
+        {
+          accessor.set_dof_index(d, index_value, fe_index);
+        }
+
+        /**
+         * Fallback for DoFInvalidAccessor.
+         */
+        template <int structdim_>
+        static DEAL_II_ALWAYS_INLINE void
+        process_dof(
+          const dealii::DoFInvalidAccessor<structdim_,
+                                           DoFHandlerType::dimension,
+                                           DoFHandlerType::space_dimension> &,
+          const unsigned int,
+          const types::global_dof_index &,
+          const unsigned int)
+        {
+          Assert(false, ExcInternalError());
+        }
+      };
+
+
+
+      template <typename DoFHandlerType, bool level_dof_access, int structdim>
+      static void
+      get_dof_indices(
+        const dealii::DoFAccessor<structdim, DoFHandlerType, level_dof_access>
+          &                                   accessor,
+        std::vector<types::global_dof_index> &dof_indices,
+        const unsigned int                    fe_index)
+      {
+        process_dof_indices(
+          accessor,
+          dof_indices,
+          fe_index,
+          DoFIndexGetter<DoFHandlerType, level_dof_access, structdim>());
+      }
+
+
+
+      template <typename DoFHandlerType, bool level_dof_access, int structdim>
+      static void
+      set_dof_indices(
+        const dealii::DoFAccessor<structdim, DoFHandlerType, level_dof_access>
+          &                                         accessor,
+        const std::vector<types::global_dof_index> &dof_indices,
+        const unsigned int                          fe_index)
+      {
+        // Note: this function is as general as `get_dof_indices()`. This
+        // assert is placed here since it is currently only used by the
+        // function DoFCellAccessor::set_dof_indices(), which is called by
+        // internal::DoFHandlerImplementation::Policy::Implementation::distribute_dofs().
+        // In the case of new use cases, this assert can be removed.
```

Comment:
Then what's the point of this assert, if the function always works?

Reply:
> Then what's the point of this assert, if the function always works?

We setting dofs till now only cell-wise. As long as we don't need s.th. more sophisticated I would like to guarantee that this is done also in the (near) future. My intention with this PR was mainly to merge the logic of 4 functions and simplify it and not add functions, which can be accessed by the user.

> This needs a test.

What kind of tests are you thinking about? These are internal implementations. The functions are already tested by all tests, which use the `DoFHandler`.